File: ce/customer-engagement/on-premises/snippets/csharp/CRMV8/optionsets/cs/workwithglobaloptionsets7.cs
// <snippetworkwithglobaloptionsets7>


 // In order to change labels on option set values (or delete) option set
 // values, you must use UpdateOptionValueRequest 
 // (or DeleteOptionValueRequest).
 UpdateOptionValueRequest updateOptionValueRequest =
     new UpdateOptionValueRequest
     {
         OptionSetName = _globalOptionSetName,
         // Update the second option value.
         Value = optionList[1].Value.Value,
         Label = new Label("Updated Option 1", _languageCode)
     };

 _serviceProxy.Execute(updateOptionValueRequest);

 //Publish the OptionSet
 PublishXmlRequest pxReq3 = new PublishXmlRequest { ParameterXml = String.Format("<importexportxml><optionsets><optionset>{0}</optionset></optionsets></importexportxml>", _globalOptionSetName) };
 _serviceProxy.Execute(pxReq3);

// </snippetworkwithglobaloptionsets7>

File: ce/customer-engagement/on-premises/snippets/visualbasic/CRMV8/optionsetsvb/vb/workwithglobaloptionsets7.vb
' <snippetworkwithglobaloptionsets7>


 ' In order to change labels on option set values (or delete) option set
 ' values, you must use UpdateOptionValueRequest 
 ' (or DeleteOptionValueRequest).
 Dim updateOptionValueRequest As UpdateOptionValueRequest =
  New UpdateOptionValueRequest With {
   .OptionSetName = _globalOptionSetName,
   .Value = optionList(1).Value.Value,
   .Label = New Label("Updated Option 1", _languageCode)
  }
 ' Update the second option value.

 _serviceProxy.Execute(updateOptionValueRequest)

 'Publish the OptionSet
 Dim pxReq3 As PublishXmlRequest =
  New PublishXmlRequest With {
   .ParameterXml =
   String.Format("<importexportxml><optionsets><optionset>{0}</optionset></optionsets></importexportxml>",
                 _globalOptionSetName)
  }
 _serviceProxy.Execute(pxReq3)

' </snippetworkwithglobaloptionsets7>